lang/compiler.go

```diff
@@ -64,14 +64,27 @@ func (c FluxCompiler) CompilerType() flux.CompilerType {
 	return FluxCompilerType
 }
 
-// SpecCompiler implements Compiler by returning a known spec.
-type SpecCompiler struct {
+// ReplCompiler is what was formerly known to be a SpecCompiler.
+// It was explicitly renamed to denote its use in the Flux REPL and only the Flux REPL.
```

Comment:
_nit_ Coding convention in Go is to use call caps for abbreviations, so this would be `REPLCompiler`.

https://github.com/golang/go/wiki/CodeReviewComments#initialisms